src/date.c

```diff
@@ -1,18 +1,1006 @@
 /*
```

Comment:
Our codebase tries to follow http://google-styleguide.googlecode.com/svn/trunk/cppguide.html
and for the optional parts we just favour consistency.

I'll focus on other stuff and we'll fix the style later.